frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailsColumns.tsx

```diff
@@ -0,0 +1,219 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { Badge, LabelGroup, Skeleton } from '@patternfly/react-core'
+import { ListItems } from '../../../../ui-components'
+import { generatePath, Link } from 'react-router-dom-v5-compat'
+import { NavigationPath } from '../../../../NavigationPath'
+import { ExternalLinkAltIcon } from '@patternfly/react-icons'
+import { TFunction } from 'react-i18next'
+import { ReactNode } from 'react'
+import { collectKinds } from '../../common/util'
+import { Grid as MuiGrid } from '@mui/material'
+
+interface IKinds {
+  apiGroups: string[]
+  kinds: string[]
+}
+// Any resources that are related to Vapb then add these rows
+export const addRowsForHasVapb = (
+  cols: ListItems[],
+  hasVapb: boolean,
+  loading: boolean,
+  vapbItems: any[] | null | undefined,
+  apiGroup: string,
+  clusterName: string,
+  name: string
+): ListItems[] => {
+  if (hasVapb) {
+    // Loading to fetch VAPB
+    if (!vapbItems) {
+      cols.push({
+        key: 'Validating Admission Policy Binding',
+        value: <Skeleton width="100%" screenreaderText="Fetching ValidatingAdmissionPolicyBinding" />,
+      })
+    } else if (vapbItems && vapbItems.length > 0 && !loading) {
+      const vapbName = apiGroup === 'constraints.gatekeeper.sh' ? `gatekeeper-${name}` : name + '-binding'
+      cols.push({
+        key: 'Validating Admission Policy Binding',
+        value: (
+          <Link
+            to={generatePath(NavigationPath.discoveredPolicyDetails, {
+              clusterName,
+              apiVersion: 'v1',
+              apiGroup: 'admissionregistration.k8s.io',
+              kind: 'ValidatingAdmissionPolicyBinding',
+              templateName: vapbName,
+              templateNamespace: null,
+            })}
+            target="_blank"
+          >
+            {vapbName} <ExternalLinkAltIcon style={{ verticalAlign: '-0.125em', marginLeft: '8px' }} />
+          </Link>
+        ),
+      })
+    } else if (vapbItems && vapbItems.length == 0 && !loading) {
+      cols.push({
+        key: 'Validating Admission Policy Binding',
+        value: '-',
+      })
+    }
+  }
+
+  return cols
+}
+
+export const addRowsForVapb = (cols: ListItems[], template: any, clusterName: string, kind: string): ListItems[] => {
+  if (kind === 'ValidatingAdmissionPolicyBinding') {
```

Comment:
Optional: In these cases, it's nice to just return `cols` if the `kind` is not equal to `ValidatingAdmissionPolicyBinding` to avoid the indentation below.